src/main.rs

```diff
@@ -14,13 +15,46 @@ fn main() -> Result<(), Box<dyn Error>> {
         process::exit(0);
     }
 
-    let warnings = dotenv_linter::run(args, &current_dir)?;
+    let warnings = dotenv_linter::run(&args, &current_dir)?;
     if warnings.is_empty() {
         process::exit(0);
     }
 
-    warnings.iter().for_each(|w| println!("{}", w));
-    process::exit(1);
+    print_warnings(args.is_present("no-color"), warnings);
+    process::exit(1)
+}
+
+fn print_warnings(no_color: bool, warnings: Vec<dotenv_linter::common::Warning>) {
+    if no_color {
+        warnings.iter().for_each(|w| {
+            println!(
+                "{}:{} {}: {}",
+                w.line.file.path.display(),
+                w.line.number,
+                w.check_name,
+                w.message
+            )
+        });
+        println!("{} problems", warnings.len());
```

Comment:
I doubt we need print warning count as least for `no-color`output :thinking: 
This may be redundant for CI systems and etc.